drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -353,6 +356,36 @@ static void delay_us(unsigned us)
 #endif
 }
 
+static inline void wait_for_write_enable_cleared(const mtd_spi_nor_t *dev)
+{
+    do {
+        uint8_t status;
+        mtd_spi_cmd_read(dev, dev->params->opcode->rdsr, &status, sizeof(status));
+
+        TRACE("mtd_spi_nor: wait device status = 0x%02x, waiting !WEL-Flag\n",
+              (unsigned int)status);
+        if ((status & SPI_NOR_STATUS_WEL) == 0) {
+            break;
+        }
+        thread_yield();
```

Comment:
Waiting for WEL usually only takes one cycle, because the majority of the waiting is done in the "wait_for_write_complete" function. The WIP and WEL flags are usually cleared together on a successful write.

Waiting for a write to complete depends a lot on the operation. For short writes it cycles only once, but a chip erase can take up to 400 seconds (6.5 minutes) for my big 1GBit Flash. But that waiting is not done in the wait_for_write_enable_cleared function.